clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -798,7 +798,7 @@ public void recordTopicFetchMetrics(String topic, int bytes, int records) {
             Sensor recordsFetched = this.metrics.getSensor(name);
             if (recordsFetched == null) {
                 Map<String, String> metricTags = new HashMap<>(1);
-                metricTags.put("topic", topic.replace('.', '_'));
+                metricTags.put("topic", topic);
```

Comment:
I agree. I will revert in this PR. And when we change it in a separate PR we will also add explanations in upgrade docs.